homeassistant/components/canary/camera.py

```diff
@@ -2,54 +2,67 @@
 import asyncio
 from datetime import timedelta
 import logging
+from typing import Callable, List
 
 from haffmpeg.camera import CameraMjpeg
 from haffmpeg.tools import IMAGE_JPEG, ImageFrame
 import voluptuous as vol
 
 from homeassistant.components.camera import PLATFORM_SCHEMA, Camera
 from homeassistant.components.ffmpeg import DATA_FFMPEG
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.helpers import config_validation as cv
 from homeassistant.helpers.aiohttp_client import async_aiohttp_proxy_stream
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
 from homeassistant.util import Throttle
 
-from . import DATA_CANARY, DEFAULT_TIMEOUT
+from . import CanaryData
+from .const import (
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_CANARY,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_FFMPEG_ARGUMENTS = "ffmpeg_arguments"
-DEFAULT_ARGUMENTS = "-pred 1"
-
 MIN_TIME_BETWEEN_SESSION_RENEW = timedelta(seconds=90)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_FFMPEG_ARGUMENTS, default=DEFAULT_ARGUMENTS): cv.string}
+    {vol.Optional(CONF_FFMPEG_ARGUMENTS, default=DEFAULT_FFMPEG_ARGUMENTS): cv.string}
```

Comment:
We don't support mixing platform config sections with config entries. The platform config sections should be deprecated when introducing config entries for an integration.

Reply:
ah ok i thought this required to ensure it parsed for importing the setting

Reply:
deprecation added here. https://github.com/home-assistant/core/pull/40355